Tribler/Test/Core/test_threadpoolmanager.py

```diff
@@ -0,0 +1,45 @@
+from time import sleep
+from Tribler.Core.APIImplementation.threadpoolmanager import ThreadPoolManager
+from Tribler.Test.Core.base_test import TriblerCoreTest
+
+
+class TriblerCoreTestThreadpoolManager(TriblerCoreTest):
+
+    def setUp(self):
+        super(TriblerCoreTestThreadpoolManager, self).setUp()
+        self.tpm = ThreadPoolManager()
+        self.callback_called = False
+
+    def callback_func(self):
+        self.callback_called = True
+
+    def test_check_task_name(self):
+        self.assertEqual(self.tpm._check_task_name("test"), "test")
+        self.assertEqual(self.tpm._check_task_name(None), "threadpool_manager 1")
+
+    def test_add_task(self):
+        self.tpm.add_task(self.callback_func, 0.0)
+        sleep(0.1)
```

Comment:
Use a deferred with a timeout decorator, let's try not to add more sleeps if we can avoid it.